chain/network-primitives/README.md

```diff
@@ -0,0 +1,33 @@
+# `near-network-primitives`
+
+`near-network-primitives` is a package that contains types, which contains network types that are meant to be shared
+between multiple crates.
+
+We have a few use cases:
+- Messages used by `ClientActor`, `ViewClientActor`, `near-network`.
+- external packages, which want to analyze traffic, they don't need to import `near-network`, importing just network types should be enough for them
```

Comment:
This feels like two very different use-cases, which ideally should be served by different crate. We definitely should have a separate crate that defines the physical "wire format" used by our p2p protocol, but that crate shouldn't contain any other impl details of nearcore (eg, the fact that we use actors in our impl shouldn't have any bearing on the actual protocol).

Reply:
> This feels like two very different use-cases, which ideally should be served by different crate.
> We definitely should have a separate crate that defines the physical "wire format" used by our p2p protocol,

@matklad Yes, exactly. I see you noticed my goal.

My plan has a few steps:
- First.  split `types.rs`, and move `network protocol` related types to `network_protocol.rs`. See https://github.com/near/nearcore/issues/5653
That's obviously, good, easy to do.
- Then I was going to propose, to move p2p network related types out of `near-network` to `near-network-primitives`. That way you would be able to use, `p2p netowrk` types by only importing `near-network-primitives` and without having to import `near-network`

Btw, the types that implement `actix` `Message`, only make sense to import, if you are going to use them to communicate with `PeerManagerActor`. I would argue that moving those types to `near-network` makes sense, because you have to import `near-network` anyway, to have to use it.

Right now, the split is kind of pointless, if you want to import types from `near-network-primitives` and use them, you have to import `near-network` as well.